packages/account-v2/src/modules/IDVForm/index.tsx

```diff
@@ -0,0 +1,159 @@
+
+    const IDV_NOT_APPLICABLE_OPTION = useMemo(() => getIDVNotApplicableOption(allowIDVSkip), [allowIDVSkip]);
+
+    const bindDocumentData = (item: string) => {
+        setFieldValue('document_type', getSelectedDocumentConfigData(item), true);
+        if (item === IDV_NOT_APPLICABLE_OPTION.value) {
+            setFieldValue('document_number', '', true);
+            setFieldValue('document_additional', '', true);
+        }
+    };
+
+    const handleSelect = (item: string) => {
+        if (item === 'No results found') {
+            setFieldValue('document_type', defaultDocument, true);
+        } else {
+            bindDocumentData(item);
+        }
+    };
+
+    const getSelectedDocumentConfigData = useCallback((item: string) => {
+        return DOCUMENT_LIST.find(doc => doc.id === item);
+    }, []);
+
+    useEffect(() => {
+        if (Object.keys(documents_supported as Exclude<typeof documents_supported, undefined>)?.length) {
+            const docList = Object.keys(documents_supported as Exclude<typeof documents_supported, undefined>).map(
+                (key: string) => {
+                    return {
+                        text: (documents_supported as Exclude<typeof documents_supported, undefined>)[key].display_name,
+                        value: key,
+                    };
+                }
+            );
+            setDocumentList(docList as TDropDownList[]);
+        }
+    }, [documents_supported]);
+
+    return (
+        <Fragment>
+            <section>
+                <Field name='document_type'>
+                    {({ field }: FieldProps) => (
+                        <ResponsiveWrapper>
+                            {{
+                                desktop: (
+                                    <WalletDropdown
+                                        {...field}
+                                        errorMessage='Document type is required'
+                                        isRequired
+                                        label='Choose the document type'
+                                        list={documentList}
+                                        onChange={handleChange}
+                                        onSelect={handleSelect}
+                                        variant='comboBox'
+                                    />
+                                ),
+                                mobile: (
+                                    <WalletDropdown
+                                        {...field}
+                                        errorMessage='Document type is required'
+                                        isRequired
+                                        label='Choose the document type'
+                                        list={documentList}
+                                        onChange={handleChange}
+                                        onSelect={handleSelect}
```

Comment:
Why do we have `onChange` and `onSelect` in a single component. Is there a specific reason to use both?

Reply:
Let me check on this. They are required by the custom component being used. Let me remove onChange and see if it works